### PR TITLE
Jr2 156 canton autocomplete

### PR DESCRIPTION
**Pull request by michael-wirth**

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchIndexServiceImpl.java [hunk 1]
```diff
@@ -39,16 +44,18 @@
...
     private final EntityToSynonymMapper entityToSynonymMapper;
 
     public ElasticsearchIndexServiceImpl(
-        ClassificationRepository classificationRepository,
-        ClassificationSearchRepository classificationSearchRepository,
-        OccupationSynonymRepository occupationSynonymRepository,
-        OccupationSynonymSearchRepository occupationSynonymSearchRepository,
-        LocalityRepository localityRepository,
-        LocalitySynonymSearchRepository localitySynonymSearchRepository,
+        ClassificationRepository classificationRepository, ClassificationSearchRepository classificationSearchRepository,
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchIndexServiceImpl.java]: I think in future we need to revisit this class and reduce number of constructor

File: src/main/java/ch/admin/seco/service/reference/service/impl/LocalityServiceImpl.java [hunk 2]
```diff
@@ -120,19 +123,40 @@ public void delete(UUID id) {
...
+            .distinct() // eliminate duplicates as 'Zürich'
+            .limit(resultSize) // reduce the result list to the desired result size
+            .collect(toList());
+
+
+        List<CompletionSuggestion.Entry> cantonResults = new ArrayList<>();
+        cantonResults.addAll(searchResponse.getSuggest().<CompletionSuggestion>getSuggestion("cantonCodes").getEntries());
+        cantonResults.addAll(searchResponse.getSuggest().<CompletionSuggestion>getSuggestion("cantonNames").getEntries());
+
+        List<CantonSuggestionDto> cantons = cantonResults.stream()
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/reference/service/impl/LocalityServiceImpl.java]: We can extract method this expression to method that accepts list and mapper function.

File: src/main/java/ch/admin/seco/service/reference/repository/CantonRepository.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import ch.admin.seco.service.reference.domain.Canton;
+
+
+/**
+ * Spring Data JPA repository for the Locality entity.
+ */
+@SuppressWarnings("unused")
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/reference/repository/CantonRepository.java]: This annotation could be removed as service used in ES index service

File: src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchIndexServiceImpl.java (hunk 1 above)
Comment (michael-wirth) [reply src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchIndexServiceImpl.java]: good point!

File: src/main/java/ch/admin/seco/service/reference/repository/CantonRepository.java (hunk 3 above)
Comment (michael-wirth) [reply src/main/java/ch/admin/seco/service/reference/repository/CantonRepository.java]: done

File: src/main/java/ch/admin/seco/service/reference/service/impl/LocalityServiceImpl.java (hunk 2 above)
Comment (michael-wirth) [reply src/main/java/ch/admin/seco/service/reference/service/impl/LocalityServiceImpl.java]: done

**Comment by michael-wirth** [comment]
Changes commited. Unfortunately I used 'amend' to instead of a clean commit. sorry for that

**Pull request merged**